tests/networkpolicy_test.go

```diff
@@ -176,64 +239,150 @@ func skipNetworkPolicyRunningOnKindInfra() {
 	}
 }
 
-func createVMICirros(virtClient kubecli.KubevirtClient, namespace string, labels map[string]string) *v1.VirtualMachineInstance {
+func createVMICirros(virtClient kubecli.KubevirtClient, namespace string, labels map[string]string, opts ...libvmi.Option) *v1.VirtualMachineInstance {
 	var err error
 	vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
 	vmi.Namespace = namespace
 	vmi.Labels = labels
+
+	// Clean up interfaces since we configure them with `libvmi.Option`
+	vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{}
+
+	for _, opt := range opts {
+		opt(vmi)
+	}
+
 	vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
 	Expect(err).ToNot(HaveOccurred())
 
 	return vmi
 }
 
-func assertPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance) {
-	ExpectWithOffset(1, toVmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
-	ConsistentlyWithOffset(1, func() error {
-		for _, toIp := range toVmi.Status.Interfaces[0].IPs {
-			if err := tests.PingFromVMConsole(fromVmi, toIp); err != nil {
-				return err
-			}
-		}
-		return nil
-	}, 15*time.Second, 1*time.Second).Should(Succeed())
+func assertICMPPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance) {
+	assertIPConnectivitySucceed(checkIPsFromVMI(toVmi, checkICMPPingAndStopOnFailure(fromVmi)))
```

Comment:
I like how this composes, but, unless I'm missing something, it can be simplified; the input function to `assertIPConnectivitySucceed` is always the same - `checkIPsFromVMI` .

I think you could merge those 2, something like:

```golang
func checkIPsFromVMI(vmi *v1.VirtualMachineInstance, checkIp func(ip string) (bool, error)) func() error {
	return func() error {
		var err error
		for _, ip := range vmi.Status.Interfaces[0].IPs {
			var shouldStop bool
			shouldStop, err = checkIp(ip)
			if shouldStop {
				return err
			}
		}
		return err
	}
}
func assertIPConnectivitySucceed(vmi *v1.VirtualMachineInstance,  checkIp func(ip string) (bool, error)) {
	ConsistentlyWithOffset(1, checkIPsFromVMI(vmi, checkIp), 15*time.Second, 1*time.Second).Should(Succeed())
}
```

You could then invoke it like:
```golang
assertIPConnectivitySucceed(vmi, checkICMPPingAndStopOnFailure(fromVmi))
```